ts/src/index.ts

```diff
@@ -90,23 +101,37 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
                     // ignore errors, which will occur when not on GCE.
                 }) ||
         [undefined, undefined];
-    if (!mergedConfig.zone && zone) {
-      mergedConfig.zone = zone.substring(zone.lastIndexOf('/') + 1);
+    if (!config.zone && zone) {
+      config.zone = zone.substring(zone.lastIndexOf('/') + 1);
     }
-    if (!mergedConfig.instance && instance) {
-      mergedConfig.instance = instance;
+    if (!config.instance && instance) {
+      config.instance = instance;
     }
   }
-
-  if (!hasService(mergedConfig)) {
-    throw new Error('Service must be specified in the configuration.');
-  }
-
-  return mergedConfig;
+  return config;
 }
 
 let profiler: Profiler|undefined = undefined;
 
+/**
+ * Initializes the config, and starts heap profiler if the heap profiler is
+ * needed.
+ */
+export async function startHelper(config: Config): Promise<ProfilerConfig> {
+  let profilerConfig = initConfig(config);
+
+  // Start the heap profiler if profiler config does not indicate heap profiling
+  // is disabled. This must be done before any asynchronous calls are made so
+  // all memory allocations made after start() is called can be captured.
+  if (!config.disableHeap) {
```

Comment:
Is this intentional that this checks config and not profilerConfig? If so, why it's not before the initConfig() call?

Reply:
Fixed.

Reply:
Not fixed?